src/Layout.js

```diff
@@ -654,9 +663,112 @@ var computeLayout = (function() {
 
       linesCrossDim += crossDim;
       linesMainDim = fmaxf(linesMainDim, mainDim);
+      linesCount += 1;
       startLine = endLine;
     }
 
+    // <Loop DD>
+    //
+    // PIERRE: More than one line, we need to layout the crossAxis according to
+    // alignContent.
+    //
+    // Note that we could probably remove <Loop D> and handle the one line case
+    // here too, but for the moment this is safer since it won't interfere with
+    // previously working code.
+    //
+    // See specs:
+    // http://www.w3.org/TR/2012/CR-css3-flexbox-20120918/#layout-algorithm
+    // section 9.4
+    //
+    if (linesCount > 1 &&
+        (!isUndefined(node.layout[dim[crossAxis]])))
+    {
+      var/*float*/ nodeCrossAxisInnerSize = node.layout[dim[crossAxis]] -
+          getPaddingAndBorderAxis(node, crossAxis);
+      var/*float*/ remainingCrossDim = nodeCrossAxisInnerSize - linesCrossDim;
+
+      var/*float*/ crossDimAdd = 0;
+      var/*float*/ currentLead = getPaddingAndBorder(node, leading[crossAxis]);
+
+      var/*css_align_t*/ alignContent = getAlignContent(node);
+      if (alignContent == CSS_ALIGN_FLEX_END) {
+        currentLead += remainingCrossDim;
+      }
+      else if (alignContent == CSS_ALIGN_CENTER) {
+        currentLead += remainingCrossDim / 2;
+      }
+      else if (alignContent == CSS_ALIGN_STRETCH) {
+        if (nodeCrossAxisInnerSize > linesCrossDim) {
+          crossDimAdd = (remainingCrossDim / linesCount);
+        }
+      }
+
+      // find the first node on the first line
+      for (i = 0; i < node.children.length; ) {
+        var/*int*/ startIndex = i;
+        var/*int*/ lineIndex = -1;
+
+        // get the first child on the current line
+        {
+          child = node.children[i];
+          if (getPositionType(child) != CSS_POSITION_RELATIVE) {
+            ++i;
+            continue;
+          }
+          lineIndex = child.lineIndex;
+        }
+
+        // compute the line's height and find the endIndex
+        var/*float*/ lineHeight = 0;
+        for (ii = startIndex; ii < node.children.length; ++ii) {
+          child = node.children[ii];
+          if (getPositionType(child) != CSS_POSITION_RELATIVE) {
+            continue;
+          }
+          if (child.lineIndex != lineIndex) {
+            break;
+          }
+          if (!isUndefined(child.layout[dim[crossAxis]])) {
```

Comment:
what if none of the children have a specified dimension? Is it properly handled?